### ISSUE TITLE
Version 1.7.1 can’t handle some paths in Nova 9.x

### ISSUE THREAD

**Issue by kopischke** [apexskier/nova-eslint#277]
PR #276 addressed changes of the API contract for file paths with Nova’s update to version 9. Unluckily, due to  faulty analysis of the cause underlying these changes, the resulting patch is bound to cause problems down the road. What Nova 9 did change is not URI handling as such. What changed is that [absolute paths returned by the Nova API do not start at the notional volume mount point (“/Volumes/Macintosh HD” by default) anymore, but at the \*nix root](https://devforum.nova.app/t/suggestions-for-converting-workspace-path/465/16). This also translates to document URIs.

The problem with #276 is that, probably due to this faulty analysis, it throws the baby out with the bathwater by removing the URI decoding step. Document URIs are still URIs in Nova 9, i.e. they still do encode non URI fragment characters like spaces and, without the decode step, version 1.7.1 will fail on all paths containing such characters.

PS: good to see you are still around, @apexskier!

**Comment by kopischke** [comment]
**EDIT:** this is wrong, see below: ~~FYI, Nova 9.1. rolled back the relevant contract change; path expansion functions return paths including the notional root volume mount point again. This includes document URIs, meaning that I’d expect **all** paths to be broken in 1.7.1.~~

**Comment by apexskier** [comment]
Good to know, thanks for the update. I actually haven't paid for the most recent updates to Nova, so I wasn't able to test the last patch directly, but there's a sale going on so I'll probably re-up my subscription.

For now, I'm going to revert the changes from https://github.com/apexskier/nova-eslint/pull/276 to get to a steady state.

**Comment by apexskier** [comment]
I've been on sabbatical from real-life work and have had non-programming projects taking up my time, so not much activity from me on GitHub.

**Comment by kopischke** [comment]
That sounds like a really healthy thing to do! The little Novaverse can wait for you, and in case you decide you’d rather not come back to it at all, feel free to holler: I have been fishing in your pond for a while, admittedly with a microscopic amount of success, but I’d still be willing and try to uphold the torch.

**Comment by gnattu** [comment]
I submitted an PR based on @kopischke 's implementation to fix this issue and it is working fine for me.

**Comment by kopischke** [comment]
@apexskier I am sorry to say I may have contributed to the broken present state of the extension: **contrary to what I stated, Nova 9.1. did not actually roll back the contract change for path expansion functions and the document path / URI handling**. Testing on 9.1 shows that

```js
nova.path.normalize('/')  // => /
nova.path.expanduser('~') // => /Users/user
TextEditor.document.path  // => /some/path/to/doc
TextEditor.document.uri   // => file:///some/path/to/doc
```

I could have _sworn_ I saw the “/Volumes Macintosh HD” part pop up back on 9.1, but whatever I saw and said is wrong, as my tests show.

My diagnosis of why rolling the patch back didn’t solve the issue, e.g. that `decodeURI()` now removes “file://” prefixes – https://github.com/apexskier/nova-eslint/issues/282#issuecomment-1078889213)https://github.com/apexskier/nova-eslint/issues/282#issuecomment-1078889213, is also wrong (sigh; I am not having a good day).

```js
decodeURI(nova.workspace.activeTextEditor.document.uri)  // => file:///some/path/to/doc
```

I am not sure why, in this light, rolling back the patch broke things and, seeing that I seem to add more confusion than light to the issue(s), I will refrain from further guesses.

**Comment by kopischke** [comment]
> I am not sure why, in this light, rolling back the patch broke things and, seeing that I seem to add more confusion than light to the issue(s), I will refrain from further guesses.

Well, obviously it breaks things because it now removes two path elements too many (the former “Volumes” and “\<volname\>” parts). Shows how hard it is to track back on a wrong train of thought and why I really, _really_ should check my assumptions better before posting. I am frightfully sorry for all of this.

**Comment by apexskier** [comment]
Included in release [v1.7.3](https://github.com/apexskier/nova-eslint/releases/tag/v1.7.3)
